### ISSUE TITLE
Errors during high update frequency (getRecord/discard).

### ISSUE THREAD

**Issue by ronag** [deepstreamIO/deepstream.io-client-js#180]
This is why I believe discard should have a timeout/throttle in case the client quickly unsubscribed and subscribes to the same record. 

In particular I think discard should have a ~2000ms timeout before it actually sends `C.ACTIONS.UNSUBSCRIBE` and if it is sending `C.ACTIONS.UNSUBSCRIBE` it should wait for an ack before it sends `C.ACTIONS.SUBSCRIBE`again.

Sample:

```javascript
import deepstream from 'deepstream.io-client-js'

const ds = deepstream('0.0.0.0:6021')
  .login({}, (success, authData) => {
    if (!success) {
      console.log({ authData }, 'Deepstream Authentication Failed. Shutting down...')
      setTimeout(() => process.exit(1), 500)
    }
  })

ds.on('connectionStateChanged', connectionState => {
  console.log({ connectionState }, 'Deepstream Connection State Changed.')
})

ds.on('error', (error, event, topic) => {
  console.log({ error, event, topic }, 'Deepstream Error.')
})

setTimeout(() => {

  setInterval(() => {
    // for (let n = 0; n < 8; ++n) {
      updateRecord()
    // }
  }, 1)

  let x = 0
  function updateRecord () {
    const record = ds.record.getRecord('test/1')
    record.set({ x })
    record.discard()
    x++
  }

}, 1000)
```

Output:

```
{ connectionState: 'AWAITING_CONNECTION' } 'Deepstream Connection State Changed.'
{ connectionState: 'AWAITING_AUTHENTICATION' } 'Deepstream Connection State Changed.'
{ connectionState: 'AUTHENTICATING' } 'Deepstream Connection State Changed.'
{ connectionState: 'OPEN' } 'Deepstream Connection State Changed.'
{ error: 'test/1', event: 'UNSOLICITED_MESSAGE', topic: 'R' } 'Deepstream Error.'
{ error: 'test/1', event: 'MULTIPLE_SUBSCRIPTIONS', topic: 'R' } 'Deepstream Error.'
{ error: 'test/1', event: 'NOT_SUBSCRIBED', topic: 'R' } 'Deepstream Error.'
{ error: 'test/1', event: 'MULTIPLE_SUBSCRIPTIONS', topic: 'R' } 'Deepstream Error.'
{ error: 'test/1', event: 'ACK_TIMEOUT', topic: 'R' } 'Deepstream Error.'
{ error: 'test/1', event: 'RESPONSE_TIMEOUT', topic: 'R' } 'Deepstream Error.'
{ error: 'test/1', event: 'RESPONSE_TIMEOUT', topic: 'R' } 'Deepstream Error.'
```

**Comment by yasserf** [comment]
This should be fixed in release 1.0.2

Having a discard timeout is another feature, but we would need to rename this issue and put it down as an enhancement rather than a bug

**Comment by ronag** [comment]
Duplicate of #204
